test/libraries/CallLib.t.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.23;
+
+import {Test} from "forge-std/Test.sol";
+import {Call, CallLib} from "../../src/libraries/CallLib.sol";
+
+contract CallLibTest is Test {
+    /// @notice Test to catch accidental changes to the typehash
+    function test_constant_typehash() public pure {
+        bytes32 expectedTypeHash = keccak256("Call(address to,uint256 value,bytes data)");
+        assertEq(CallLib.CALL_TYPEHASH, expectedTypeHash);
+    }
+
+    function test_hash_single_fuzz(address to, uint256 value, bytes calldata data) public pure {
+        Call memory call = Call({to: to, value: value, data: data});
+        bytes32 actualHash = CallLib.hash(call);
+
+        bytes32 expectedHash = keccak256(abi.encode(CallLib.CALL_TYPEHASH, call.to, call.value, call.data));
+        assertEq(actualHash, expectedHash);
+    }
+
+    function test_hash_multiple_fuzz(Call[] memory calls) public pure {
+        bytes32 actualHash = CallLib.hash(calls);
+
+        bytes memory packedHashes = new bytes(32 * calls.length);
+        for (uint256 i = 0; i < calls.length; i++) {
+            bytes32 callHash = CallLib.hash(calls[i]);
+            assembly {
+                mstore(add(add(packedHashes, 0x20), mul(i, 0x20)), callHash)
+            }
+        }
```

Comment:
would do just a vanilla abi.encodePacked() so its comparing the assembly to something non-assembly